src/helics/core/CoreBroker.cpp

```diff
@@ -2583,6 +2683,40 @@ void CoreBroker::initializeDataFlowGraph()
     }
 }
 
+void CoreBroker::initializeCurrentTimeMap()
+{
+    Json::Value& base = currentTimeMap.getJValue();
+    base["name"] = getIdentifier();
+    base["id"] = global_broker_id_local.baseValue();
+    if (!isRootc) {
+        base["parent"] = higher_broker_id.baseValue();
+    }
+    base["brokers"] = Json::arrayValue;
+    ActionMessage queryReq(CMD_BROKER_QUERY);
+    queryReq.payload = "global_time";
+    queryReq.source_id = global_broker_id_local;
+    queryReq.counter = 6; // indicating which processing to use
```

Comment:
Is there a processing state/type enum that could be used for this?

Reply:
That could also make it clearer what the different cases in `processQueryResponse` are for.

Reply:
I think I am going to create a story to clean up the broker queries.  When working on this I noticed a number of other areas that can be refactored to reduce the code and clean up some things.  making this number make more sense would be one of them.  But it also was not something I was going to be able to fix in a couple hours right now either.  So we could make an enumeration but it would likely just disappear once the is completed, or at the very least would be dealt with then.